internal/pkg/agent/cmd/watch.go

```diff
@@ -198,7 +198,12 @@ func watchCmd(log *logp.Logger, topDir string, cfg *configuration.UpgradeWatcher
 		// if we're not within grace and marker is still there it might mean
 		// that cleanup was not performed ok, cleanup everything except current version
 		// hash is the same as hash of agent which initiated watcher.
-		if err := installModifier.Cleanup(log, paths.Top(), paths.VersionedHome(topDir), release.ShortCommit(), true, false); err != nil {
+		versionedHomesToKeep := make([]string, 0, len(marker.RollbacksAvailable)+1)
```

Comment:
Nit: the capacity here, specifically the `len(marker.RollbacksAvailable)` part depends on `appendAvailableRollbacks` using the same length slice.  Might be safer not to make that assumption and leave the capacity as 0 (default)?

Reply:
We can leave the capacity at zero and reallocate as necessary, however the resulting slice of versioned homes to keep at the end of the grace period will always be the `<number of available rollbacks > + <the new versioned home of the upgraded elastic-agent>`